src/lib.rs

```diff
@@ -117,713 +118,660 @@ macro_rules! define_prefix_methods {
         /// Negates a number, returning None for results that can't be represented.
-        checked_neg(&self) -> (Option<Self>)
+        checked_neg(x: &T) -> Option<T>
     }
 
     /// Methods from the Inv trait, exposed as free functions
     num_traits::ops::inv::Inv => inv {
         /// Returns the multiplicative inverse of a number.
-        inv(self) -> (Self::Output)
+        inv(x: T) -> T::Output
     }
 
     /// Methods from the Real trait, exposed as free functions.
     #[cfg(any(feature = "std", feature = "libm"))]
     num_traits::real::Real => real {
         /// Returns the smallest finite value that this type can represent.
-        min_value() -> Self
+        min_value() -> T
 
         /// Returns the smallest positive, normalized value that this type can represent.
-        min_positive_value() -> Self
+        min_positive_value() -> T
 
         /// Returns epsilon, a small positive value.
-        epsilon() -> Self
+        epsilon() -> T
 
         /// Returns the largest finite value that this type can represent.
-        max_value() -> Self
+        max_value() -> T
 
         /// Returns the largest integer less than or equal to a number.
-        floor(self) -> Self
+        floor(x: T) -> T
 
         /// Returns the smallest integer greater than or equal to a number.
-        ceil(self) -> Self
+        ceil(x: T) -> T
 
         /// Returns the nearest integer to a number.
-        round(self) -> Self
+        round(x: T) -> T
 
         /// Returns the integer part of a number.
-        trunc(self) -> Self
+        trunc(x: T) -> T
 
         /// Returns the fractional part of a number.
-        fract(self) -> Self
+        fract(x: T) -> T
 
         /// Computes the absolute value of a number.
-        abs(self) -> Self
+        abs(x: T) -> T
 
         /// Returns a number that represents the sign of the input.
-        signum(self) -> Self
+        signum(x: T) -> T
 
         /// Returns true if the input is positive.
-        is_sign_positive(self) -> bool
+        is_sign_positive(x: T) -> bool
 
         /// Returns true if the input is negative.
-        is_sign_negative(self) -> bool
+        is_sign_negative(x: T) -> bool
 
         /// Fused multiply-add.
-        mul_add(self, a: Self, b: Self) -> Self
+        mul_add(x: T, a: T, b: T) -> T
 
         /// Take the reciprocal (inverse) of a number, `1/x`.
-        recip(self) -> Self
+        recip(x: T) -> T
 
         /// Raise a number to an integer power.
-        powi(self, n: i32) -> Self
+        powi(x: T, n: i32) -> T
 
         /// Raise a number to a real number power.
-        powf(self, n: Self) -> Self
+        powf(x: T, n: T) -> T
 
         /// Take the square root of a number.
-        sqrt(self) -> Self
+        sqrt(x: T) -> T
 
-        /// Returns `e^x`, (the exponential function).
-        exp(self) -> Self
+        /// Returns `e^x`, (x: The exponential function).
```

Comment:
Not sure what happened here. But anyway, the original way this was written was fishy to begin with. Here's a proposal:

```/// Returns `e^x` (the exponential function)```